mapit/models.py

```diff
@@ -265,17 +265,30 @@ def as_dict(self, all_names=None):
             'codes': self.all_codes,
             'all_names': dict(n.as_tuple() for n in all_names),
         }
-        countries = [{'code': c.code, 'name': c.name} for c in self.countries.all()]
+        countries = self.all_m2m_countries
         if countries:
             out['countries'] = countries
         return out
 
     def list_countries(self):
-        countries = [c.name for c in self.countries.all()]
+        countries = [c['name'] for c in self.all_m2m_countries]
         if self.country:
             countries.append(self.country.name)
         return countries
 
+    @property
```

Comment:
Could this use [`django.utils.functional.cached_property`](https://docs.djangoproject.com/en/2.0/ref/utils/#django.utils.functional.cached_property) instead? It'd make the code simpler and remove the need for the setter as the decorator handles that.